include/mrdox/Metadata/Record.hpp

```diff
@@ -67,12 +67,42 @@ struct BaseInfo
 */
 struct RecordScope
 {
+    // friend class ASTVisitor;
+
     std::vector<MemberRef> Records;
     std::vector<MemberRef> Functions;
     std::vector<MemberRef> Enums;
     std::vector<MemberRef> Types;
     std::vector<MemberRef> Fields;
     std::vector<MemberRef> Vars;
+
+// private:
+    template<typename T, typename... Args>
+    void emplace_back(Args&&... args)
+    {
+        if constexpr(std::is_constructible_v<MemberRef, Args...>)
+        {
+            using U = std::remove_cvref_t<T>;
+            if constexpr(std::is_same_v<U, RecordInfo>)
+                Records.emplace_back(std::forward<Args>(args)...);
+            else if constexpr(std::is_same_v<U, FunctionInfo>)
+                Functions.emplace_back(std::forward<Args>(args)...);
+            else if constexpr(std::is_same_v<U, TypedefInfo>)
+                Types.emplace_back(std::forward<Args>(args)...);
+            else if constexpr(std::is_same_v<U, EnumInfo>)
+                Enums.emplace_back(std::forward<Args>(args)...);
+            else if constexpr(std::is_same_v<U, FieldInfo>)
+                Fields.emplace_back(std::forward<Args>(args)...);
+            else if constexpr(std::is_same_v<U, VarInfo>)
+                Vars.emplace_back(std::forward<Args>(args)...);
+            else
+                llvm_unreachable("invalid RecordScope member");
```

Comment:
we need to REMOVE calls to llvm in public interfaces not ADD them.

Reply:
I was going to ask to add an `Unreachable` macro to `Platform.hpp`... so maybe I shall do that